### ISSUE TITLE
Display PDF title or first few additional keywords in documents list. [Feature request/suggestion]

### ISSUE THREAD

**Issue by eliotb** [openpaperwork/paperwork#623]
Currently in each entry in the left hand pane [Documents] there is 
- An unreadable thumbnail
- a date
- zero or more labels
- the property and delete buttons

I'd like to see the space to the right of the date occupied by the first words of Additional Keywords, or of the PDF title (if document is imported from PDF with a title).

Is this something for which a patch *might* be accepted?

**Comment by jflesch** [comment]
I tend to be against titles because the goal of Paperwork is being lazy, and entering a title is an extra (and useless) task. Therefore I don't want to encourage users to enter titles.
Displaying additional keywords in the document ;list would basically turn them into titles.

However, regarding PDF titles, I guess it could make sense since it doesn't imply extra work for the user. But I think a lot of PDF don't actually have correct markup in that regard. Still, it could make sense for the few PDF that actually have valid markup.

@mjourdan : Any thought on that one ?

**Comment by mjourdan** [comment]
There is definitely room for improvement in the left pane. Avoiding users to type in unnecessary titles is indeed a rule we should stick to, but this doesn't mean there should be no title at all. For pdf, we may have metadata to rely on, fine. But we should keep the left pane consistent, and display relevant information for scanned documents too. To do this the lazy way, instead of extra keywords we probably could extract useful information from the papers as discussed in #560:
- for mail, the subject could obviously be used as a title, the sender could also be shown, and the expedition date could also be used in place of the date of the scan
- for paychecks and bills, perhaps showing the value could also be interesting in the left pane

**Comment by eliotb** [comment]
I understand laziness :).  I didn't want to make the user enter a separate title, rather to display some data that is already present.  If a user has chosen to enter extra keywords then they would be visible.

For me, it would give a better idea of what is in the collection (without an explosion in the number of defined labels).  Again, personally, the thumbnails are too small to be useful in distinguishing between documents with same lables.

E.g. for label "insurance", keywords might be 'house', 'car', 'contents', 'medical'

**Comment by mjourdan** [comment]
I made something to illustrate what it could look like:

![image](https://raw.githubusercontent.com/mjourdan/gnome-design/master/paperwork/experiments/papers-list.png)

@eliotb, does it match your request?

**Comment by jflesch** [comment]
Hm, while this is out of the scope of this ticket, I like the idea of grouping documents by years / months :-)

However, don't remove thumbnails. OCR is hardly good enough to identify documents just from excerpt. And thumbnails makes it much more easy for the human brain to identify documents (human brain is awesome at recognizing visual patterns).

**Comment by mjourdan** [comment]
> Hm, while this is out of the scope of this ticket, I like the idea of grouping documents by years / months :-)

Yeah, I think changing the date format to use full text months could improve readability too.

> However, don't remove thumbnails. OCR is hardly good enough to identify documents just from excerpts. And thumbnails makes it much more easy for the human brain to identify documents (human brains are awesome at recognizing visual patterns).

I agree. Problem is this doesn't work when you have several documents using the same pattern even for different matters (ie, letters). So it could be nice to have some heuristics here to guess what portion of text could make a good excerpt (which relates to #560). Currently we do have free space besides the thumbnail which is worth filling in.

Anyway, that was just an attempt, numbers of the day appear to be even less meaningful than thumbnails.

**Comment by eliotb** [comment]
@mjourdan something like it.  I would prioritise my own keywords above both the doc title, and anything autoselected from the document. (because my words are likely to be most meaningful to me)
Also would place the (truncated) text in currently blank area to right of date.  Possibly with full text pop up on hover.

**Comment by mjourdan** [comment]
@eliotb, thanks for your feedback! One question comes to mind: how do you use extra keywords? I understand that you use them only to have less tags, but I can't see the point of having a few tags and a lot of keywords, rather than just having a lot of tags.

That could help in the design work we have regarding this specific feature (see #195).
